### PR TITLE
Fix #1164, #1156, #1125

### LINKED ISSUE #1164: rudimentary program doesn't type-check
A cs019 student wasted a bunch of time unable to do homework because he was unable to run

```
fun test(str :: String) -> List<Number>:
string-to-code-points(str)
end

check:
test("abc") is [list: 97, 98, 99]
end
```

(and various variants of it). The problem is that `string-to-code-points` has type `Any`, and so results in a type error. Once we got him out of the Type Checker mode, he's fine, and finally able to start on his homework.

Color me not happy. At all.

### PR DESCRIPTION
**Pull request by sorawee**

### CODE REVIEW

**Comment by mkolosick** [comment]
This should also be cherry-picked onto master so it can be released.

File: src/js/base/runtime.js [hunk 1]
```diff
@@ -5211,7 +5211,7 @@ function (Namespace, jsnums, codePoint, util, exnStackParser, loader, seedrandom
       'raw-array-from-list': makeFunction(raw_array_from_list, "raw-array-from-list"),
       'raw-array-join-str': makeFunction(raw_array_join_str, "raw-array-join-str"),
       'get-value': makeFunction(getValue, "get-value"),
-      'list-to-raw-array': makeFunction(function(l) { return thisRuntime.ffi.toArray(l); }, "list-to-raw-array"),
+      'list-to-raw-array': makeFunction(raw_array_from_list, "raw-array-from-list"),
```
Comment (blerner) [inline src/js/base/runtime.js]: I don't understand this line: I thought you were trying to get rid of this function entirely?  Why reimplement it, and change its name, if that's going to happen?

File: src/js/base/runtime.js [hunk 2]
```diff
@@ -285,7 +285,7 @@ function (Namespace, jsnums, codePoint, util, exnStackParser, loader, seedrandom
         }
         s += " }";
         return s;
-        
+
```
Comment (blerner) [inline src/js/base/runtime.js]: Realistically, all this whitespace tweaking shouldn't be part of this PR, since it just increases the likelihood of merge conflicts when we try to cherry-pick it

File: src/js/base/runtime.js (hunk 1 above)
Comment (sorawee) [reply src/js/base/runtime.js]: The plot library current uses `list-to-raw-array` because `raw-array-from-list` in `master` is not usable (buggy). Once this is merged to `master`, we can get rid of it.

**Comment by sorawee** [comment]
Really sorry for all whitespaces. I seem to have a plugin in both git and atom that make these invisible :( I will be more careful next time.
